src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -247,6 +247,10 @@ public void setTokens(Collection<Token> tokens)
 
     public void setGossipTokens(Collection<Token> tokens)
     {
+        if (DatabaseDescriptor.isAutoBootstrap() && !bootstrapComplete())
```

Comment:
Do we ever expect auto_bootstrap to be set to false?

Reply:
we never set it, but I included here to be consistent w/ the rest of the codebase